### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by hopesgit**
Adds Games, Teams, and GameTeams classes to handle their appropriate CSVs.

- Renames all of the sample CSVs (ex: games_sample.csv)
- Removed most of the content of the runner (keeping the main path/locations and the instantiated StatTracker) 
- Add Game class, which reads from the Game_sample file to create Games instances with all properties from games csv intact
- Same as above for Teams and GameTeams
- Keeps all CSV columns for now
- StatTracker can tell the other classes where to find their csv methods and stores the input in in StatTracker instance variables
- Fully tested and working

Update: Changed all files and mentions relating to Games, Teams, and GameTeams class files to drop the "s"

### CODE REVIEW

File: lib/game_team.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+              :head_coach,
+              :goals,
+              :shots,
+              :tackles,
+              :pim,
+              :ppo,
+              :ppg,
+              :fowp,
+              :giveaways,
+              :takeaways
```
Comment (jakeheft) [inline lib/game_team.rb:18]: These lines can be deleted since they relate to data we won't be using.

File: lib/game_team.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    @head_coach = data[:head_coach]
+    @goals = data[:goals]
+    @shots = data[:shots]
+    @tackles = data[:tackles]
+    @pim = data[:pim]
+    @ppo = data[:powerplayopportunities]
+    @ppg = data[:powerplaygoals]
+    @fowp = data[:faceoffwinpercentage]
+    @giveaways = data[:giveaways]
+    @takeaways = data[:takeaways]
```
Comment (jakeheft) [inline lib/game_team.rb:37]: These lines can be deleted since they relate to data we won't be using.

**Comment by jakeheft** [comment]
Correct me if I'm missing something, but it looks like your runner can no longer access sample data? Do we want to create a sample runner? Or do we not need this information in the runner file?

**Comment by dunlapww** [comment]
I think I just uncommented the regular stat tracker instance to see if my
methods worked with the real data. The print lines at the end call on stat
tracker rather than stat tracker test. But I’m fine to change back to how
it was. I really didnt mean to change runner at all.

**Comment by jakeheft** [comment]
Will, does your last comment relate to Hope's PR or yours? I'm a little confused; Hope's runner is the file I was commenting about.

**Comment by dunlapww** [comment]
Oh sorry, I’m seeing this stuff come through on gmail and prob not reading
it correctly.  Disregard my previous response. Also I had a couple PRs -
did you see the one with my methods or maybe I accidentally pushed that to
the Turing main which I do about 80% of the time....

**Comment by jakeheft** [comment]
Will, yes following through gmail is difficult - that's what I was trying to do last night and I failed miserably. I did see your 2 PRs, haven't looked too much at the methods you put up yet though. As for the stat_tracker refactor, what Hope did yesterday is different than what yours is, so we all need to figure out which direction we want to go in...

File: lib/game_team.rb (hunk 1 above)
Comment (hopesgit) [reply lib/game_team.rb:18]: Let's keep them for now. The spec harness requires a few of the columns that we won't be using for our data but are in the spec tests.

File: lib/game_team.rb (hunk 2 above)
Comment (hopesgit) [reply lib/game_team.rb:37]: Ditto for my earlier comment.

**Comment by hopesgit** [comment]
The runner will use the full data, but we don't want i t when building the methods and such because it's incredibly expensive that way. That's why y'all added the test route to the runner anyway, right?

**Review by dunlapww — APPROVED** [review]
Hi Hope, thanks again for all the work you put into this.  Unfortunately, it does look like I’ll need to recreate the methods I built to work within this new data structure.  Moving forward, I’ll try to do a better job speaking up if I’m not fully tracking on a decision—in this case, I didn’t fully understand what you were working on, when it would be completed, or how it might impact what team members were working on.  So, I’m ok with moving down this path if the rest of the team is aligned to this redesigned runner/classes.  Joshua/Jake, look forward to your perspective. Again, I appreciate the hard work Hope, thanks!

**Review by jakeheft — APPROVED** [review]

File: lib/game_team.rb (hunk 1 above)
Comment (jakeheft) [reply lib/game_team.rb:18]: Oh yeah, good call - didn't even think about it in terms of the spec harness.

**Comment by jakeheft** [comment]
My vote would be to move forward with Hope's changes as from what I have seen it seems to be a very efficient way to access the data. My one concern--and a little tweaking can solve this--is that when we change everything from our sample data to the full data almost all of the tests we build will fail, so we need to make a way to have both normal and sample data accessible inside the classes so we can build tests just for sample data. My suggestion would be add @sample_games, @sample_teams, & @sample_team_games so we explicitly know we're calling for sample data right off the bat (as opposed to @team_games_sample, which if using tab completion might drop the sample and be easily missed).
EDIT: I just realized that adding these extra class instances means changing how our methods are set up which is a definite problem. We should definitely go over this in our next meeting.

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -1,30 +1,21 @@
...
+    @game_teams = game_teams
   end
 
-  def self.from_csv(locations)
-    self.new(locations[:teams], locations[:games], locations[:game_teams])
+  def self.from_csv(locations = {games: './data/games_sample.csv', teams: './data/teams_sample.csv', game_teams: './data/game_teams_sample.csv'})
+    Game.from_csv(locations[:games])
+    Team.from_csv(locations[:teams])
+    GameTeam.from_csv(locations[:game_teams])
+    self.new(Team.all_teams, Game.all_games, GameTeam.all_game_teams)
```
Comment (jdcarey128) [inline lib/stat_tracker.rb:19]: This looks good - addresses my initial question posed in our slack channel about initiating the games, teams, and game_teams classes with the StatTracker.from_csv method.

**Pull request merged**